jsonrpc/filter_manager.go

```diff
@@ -212,6 +232,68 @@ func (f *logFilter) sendUpdates() error {
 	return nil
 }
 
+// getSubscriptionType returns the type of the event the filter is subscribed to
+func (f *logFilter) getSubscriptionType() subscriptionType {
+	return Blocks
+}
+
+// pendingTxFilter is a filter to store pending tx
+type pendingTxFilter struct {
+	filterBase
+	sync.Mutex
+
+	txHashes []string
+}
+
+// appendPendingTxHashes appends new pending tx hash to tx hashes
+func (f *pendingTxFilter) appendPendingTxHashes(txHash string) {
+	f.Lock()
+	defer f.Unlock()
+
+	f.txHashes = append(f.txHashes, txHash)
+}
+
+// takePendingTxsUpdates returns all saved pending tx hashes in filter and sets a new slice
+func (f *pendingTxFilter) takePendingTxsUpdates() []string {
+	f.Lock()
+	defer f.Unlock()
+
+	txHashes := f.txHashes
+	f.txHashes = []string{}
+
+	return txHashes
+}
+
+// getSubscriptionType returns the type of the event the filter is subscribed to
+func (f *pendingTxFilter) getSubscriptionType() subscriptionType {
+	return PendingTransactions
+}
+
+// getUpdates returns stored pending tx hashes
+func (f *pendingTxFilter) getUpdates() (interface{}, error) {
+	pendingTxHashes := f.takePendingTxsUpdates()
+
+	return pendingTxHashes, nil
+}
+
+// sendUpdates write the hashes for all pending transactions to web socket stream
+func (f *pendingTxFilter) sendUpdates() error {
+	pendingTxHashes := f.takePendingTxsUpdates()
+
+	for _, txHash := range pendingTxHashes {
+		res, err := json.Marshal(txHash)
```

Comment:
Just curious, why do we marshal a string, and then converting it back to string when we call `writeMessageToWs`?

Reply:
Yep, it seems like `txHash` could be provided to `writeMessageToWs`?